web/js/layers/wv.layers.modal.js

```diff
@@ -32,85 +32,105 @@ wv.layers.modal = wv.layers.modal || function(models, ui, config) {
   var $selectedCategory = $(self.selector + " #selected-category");
   var $allLayers = $(self.selector + " #layers-all");
   var gridItemWidth = 320; //with of grid item + spacing
+  var projection = models.proj.selected.id;
   var modalHeight;
   var sizeMultiplier;
   var searchBool;
+  var hasSetting;
 
-  // Visible Layers
+  //Visible Layers
   var visible = {};
 
   var init = function() {
     _.each(config.layers, function(layer) {
       visible[layer.id] = true;
     });
     model.events
-      // FIXME: on "add" needs to be present without trying to add a product
-      // multiple times
-      //.on("add", onLayerAdded)
+    //FIXME: on "add" needs to be present without trying to add a product
+    // multiple times
+    //.on("add", onLayerAdded)
       .on("remove", onLayerRemoved);
     models.proj.events.on("select", drawDefaultPage);
 
-    // Create tiles
+    //Create tiles
     render();
 
     $addBtn.click(function(e) {
-      $(self.selector)
-        .dialog("open");
+      $(self.selector).dialog("open");
     });
 
-    $(window)
-      .resize(resize);
+    $(window).resize(resize);
   };
 
-  // Create container for 'by interest' filters buttons
-  var $nav = $('<nav></nav>')
-    .attr('id', 'categories-nav');
+  //Create container for 'by interest' filters buttons
+  var $nav = $('<nav></nav>').attr('id', 'categories-nav');
 
-  // Create container for breadcrumb
-  var $breadcrumb = $('<nav></nav>')
-    .attr('id', 'category-breadcrumb');
+  //Create container for breadcrumb
+  var $breadcrumb = $('<nav></nav>').attr('id', 'category-breadcrumb');
+
+  var checkSettings = function(current, source) {
+    _.each( source.settings, function( setting ) {
+      var layer = config.layers[setting];
+      var proj = layer.projections;
+      if(layer.id == setting && Object.keys(proj).indexOf(projection) > -1) {
+        if (layer.layergroup && layer.layergroup.indexOf("reference_orbits") !== -1) {
+          if(current.id === "orbital-track") {
+            hasSetting = true;
+          }
+        // Don't output sources with only orbit tracks
+        } else {
+          hasSetting = true;
+        }
+      }
+    });
+  };
+
+  // Checks to see if the current view has sources with settings.
+  var checkSourceSettings = function(current) {
+    _.each( current.sources, function( source, sourceName ) {
+      checkSettings(current, source);
+    });
+  };
+
+  var checkModalView = function() {
+    if (config.parameters.modalView == 'categories') {
```

Comment:
Good opportunity for switch statement

Reply:
when using the same chained object many times within a function it can be more efficient to set it as a var at the top of the func `eg : var modalView  = config.parameters.modalview`, but if the switch was in place it wouldn't be necessary cause the chained value would only be referenced once.